### DISCUSSION TITLE
signal-desktop: tray icon does not appear

### DISCUSSION THREAD

**Discussion by rieje** [netblue30/firejail#5670] [Q&A] [answered]
Quick questions:

 signal desktop runs but does not show the icon on Wayland (it did a while ago back). This is the [output](https://0x0.st/Hs-W.txt). Any ideas?

* udiskie fails with:

    Seccomp list in: !request_key, check list: @default-keep, prelist: unknown,
    Parent pid 7259, child pid 7262
    Error fcopy: invalid ownership for file /usr/local/bin/alacritty
    Error: failed to run /run/firejail/lib/fcopy, exiting...
    Error: proc 7259 cannot sync with peer: unexpected EOF
    Peer 7262 unexpectedly exited with status 1

~~* I run `firejail /usr/bin/signal-desktop --use-tray-icon --enable-features=UseOzonePlatform --ozone-platform=wayland`~~

EDIT: Solved with `?ALLOW_TRAY: dbus-user.talk org.kde.StatusNotifierWatcher` in signal-desktop.local.

Curious if these are more common problems that have a workaround already.

**Comment by rusty-snake** [comment, marked as answer]
Did you set `allow-tray yes` in `/etc/firejail/firejail.config`?

**Reply by rieje** [reply]
No, but I did just now and restarted, still no icon and seems no difference.

**Reply by rusty-snake** [reply]
Try with this in `signal-desktop.local`:

```
?ALLOW_TRAY: dbus-user.own org.kde.*
?ALLOW_TRAY: dbus-user.talk org.kde.StatusNotifierWatcher
```

If it works, try if you can omit the first line.

**Reply by rieje** [reply]
It works with both lines but also works with second line. Does not work with first line. I will just use second line, let me know if you think there should be a change in source code off the top of your head so I can make a comment to check on it in the future (would like to stick with defaults as much as possible to keep things simple since my system setup and usage is pretty standard).

Thank you.

**Reply by rusty-snake** [reply]
Yes, it makes sense to fix this in the profile.

**Reply by rieje** [reply]
Ok, I also don't use `allow-tray yes` anymore in /etc/firejail/firejail.config because it was not necessary (in the end only `?ALLOW_TRAY: dbus-user.talk org.kde.StatusNotifierWatcher` was in signal-desktop.local).

**Reply by scopry** [reply]
for me, while using waybar, this didn't work.

for me just works:
```
dbus-user.talk org.kde.StatusNotifierWatcher
```

**Comment by rusty-snake** [comment]
> invalid ownership for file /usr/local/bin/alacritty

So how owns it?

**Reply by rieje** [reply]
/usr/local/bin/alacritty is a symlink owned by root to a file owned by me in ~/bin/default-wrapper/alacritty. I do it this way because /usr/local/bin is highest precedence in $PATH and I didn't want to add ~/bin to it.

**Comment by rusty-snake** [comment]
OT: If you have two issues/questions, it is better to open two separate Issues/Discussions.

**Reply by rieje** [reply]
Will do in the future, I only tried to add in second question because it was a low effort request for help.
